tests/unit/models/delivery-test.js

```diff
@@ -1,143 +1,119 @@
+import EmberObject from '@ember/object';
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
 import { testRelationships } from '../../helpers/test-relationships';
-import Ember from 'ember';
 
-moduleForModel('delivery', 'Unit | Model | delivery', {
-  // Specify the other units that are required for this test.
-  needs: [
-    'model:duke-ds-project',
-    'model:duke-ds-user',
-    'model:duke-ds-project-transfer'
-  ]
-});
+module('Unit | Model | delivery', function(hooks) {
+  setupTest(hooks);
+  let store, deliveryAdapter;
+  hooks.beforeEach(function() {
+    let { owner } = this;
+    store = owner.lookup('service:store');
+    deliveryAdapter = store.adapterFor('delivery');
+  });
 
-test('it exists', function(assert) {
-  let model = this.subject();
-  // let store = this.store();
-  assert.ok(!!model);
-});
+  test('it exists', async function(assert) {
+    let model = store.createRecord('delivery');
+    assert.ok(!!model);
+  });
 
-test('it defaults state to new', function(assert) {
-  let model = this.subject();
-  const STATE_NEW = 0;
-  assert.equal(model.get('state'), STATE_NEW);
-});
+  test('it defaults state to new', async function(assert) {
+    let model = store.createRecord('delivery');
+    const STATE_NEW = 0;
+    assert.equal(model.get('state'), STATE_NEW);
+  });
 
-test('it defaults userMessage to empty string', function(assert) {
-  let model = this.subject();
-  assert.equal(model.get('state'), '');
-});
+  test('it defaults userMessage to empty string', async function(assert) {
+    let model = store.createRecord('delivery');
+    assert.equal(model.get('state'), '');
+  });
 
-const relationships = [
-  {key: 'project', kind: 'belongsTo', type: 'duke-ds-project'},
-  {key: 'fromUser', kind: 'belongsTo', type: 'duke-ds-user'},
-  {key: 'toUser', kind: 'belongsTo', type: 'duke-ds-user'}
-];
+  const relationships = [
+    {key: 'project', kind: 'belongsTo', type: 'duke-ds-project'},
+    {key: 'fromUser', kind: 'belongsTo', type: 'duke-ds-user'},
+    {key: 'toUser', kind: 'belongsTo', type: 'duke-ds-user'}
+  ];
 
-testRelationships('delivery', relationships);
+  testRelationships('delivery', relationships);
 
-test('delivery.preview() calls adapter.preview()', function (assert) {
-  assert.expect(3);
-  const response = Ember.Object.create();
-  this.store().set('adapterFor', (modelName) => {
-    return {
-      preview() {
-        assert.equal(modelName, 'delivery');
-        assert.ok(true);
-        return Ember.RSVP.resolve(response);
-      }
-    }
-  });
-  let model = this.subject();
-  model.preview().then(preview => {
+  test('delivery.preview() calls adapter.preview()', async function (assert) {
+    assert.expect(2);
+    const response = EmberObject.create();
+    deliveryAdapter.preview = () => {
+          assert.ok(true);
+          return resolve(response);
+    };
+    let model = store.createRecord('delivery');
+
+    const preview = await model.preview();
```

Comment:
`await` receives a promise and pauses the code until the promise returns and returns the resulting value. In my head it is just a cleaned up version of `.then(() => ...rest_of_function...);`. 

The part I am not 100% sure on is when we would need to use the ember specific `run` method instead of `await`. The [ember-data model tests](om/emberjs/data/blob/master/packages/-ember-data/tests/unit/model-test.js) I patterned these fixes on does not use `run` at all.  The [docs for run](https://api.emberjs.com/ember/release/functions/@ember%2Frunloop/run) say 
> Runs the passed target and method inside of a RunLoop, ensuring any deferred actions including bindings and views updates are flushed at the end.

So testing dynamic changes to views will probably need the run method.

In general using `await` seems much cleaner to me.